### ISSUE TITLE
Feature: Custom import groups

### ISSUE THREAD

**Issue by azat-io** [azat-io/eslint-plugin-perfectionist#19]
### What rule do you want to change?

sort-imports

### Describe the problem

We need to add the ability to define custom import groups so that we can better sort the imports by meaning.

### Code example

```json
{
  "plugins": ["perfectionist"],
  "rules": {
    "perfectionist/sort-imports": [
      "error",
      {
        "type": "natural",
        "order": "asc",
        "custom-groups": {
          "values": {
            "react": ["react", "react-*"],
            "nanostores": ["nanostores", "@nanostores/**"]
          },
          "type": {
            "react": ["react", "react-*"]          
          },
        }
        "groups": [
          "react",
          "nanostores"
        ],
        "newlines-between": "always"
      }
    ]
  }
}
```

```js
import React from 'react'
import type { FC } from 'react'
import { render } from 'react-dom'

import { useStore } from '@nanostores/react'
```

### Validations

- [X] Read the [docs](https://eslint-plugin-perfectionist.azat.io).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by smol-honk** [comment]
Would love this!

**Comment by azat-io** [comment]
Released in [v1.3.0](https://github.com/azat-io/eslint-plugin-perfectionist/releases/tag/v1.3.0)
